airflow/providers/amazon/aws/triggers/ecs.py

```diff
@@ -0,0 +1,204 @@
+    """
+    Waits for an ECS task to be done, while eventually polling logs.
+
+    :param cluster: short name or full ARN of the cluster where the task is running.
+    :param task_arn: ARN of the task to watch.
+    :param waiter_delay: The amount of time in seconds to wait between attempts.
+    :param waiter_max_attempts: The number of times to ping for status.
+        Will fail after that many unsuccessful attempts.
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+    :param region: The AWS region where the cluster is located.
+    """
+
+    def __init__(
+        self,
+        cluster: str,
+        task_arn: str,
+        waiter_delay: int | None,
+        aws_conn_id: str | None,
+        region: str | None,
+        log_group: str | None = None,
+        log_stream: str | None = None,
+    ):
+        self.cluster = cluster
+        self.task_arn = task_arn
+
+        self.waiter_delay = waiter_delay or 15
+        self.aws_conn_id = aws_conn_id
+        self.region = region
+
+        self.log_group = log_group
+        self.log_stream = log_stream
+
+    def serialize(self) -> tuple[str, dict[str, Any]]:
+        return (
+            self.__class__.__module__ + "." + self.__class__.__qualname__,
+            {
+                "cluster": self.cluster,
+                "task_arn": self.task_arn,
+                "waiter_delay": self.waiter_delay,
+                "aws_conn_id": self.aws_conn_id,
+                "region": self.region,
+                "log_group": self.log_group,
+                "log_stream": self.log_stream,
+            },
+        )
+
+    async def run(self) -> AsyncIterator[TriggerEvent]:
+        # fmt: off
+        async with EcsHook(aws_conn_id=self.aws_conn_id, region_name=self.region).async_conn as ecs_client,\
+                AwsLogsHook(aws_conn_id=self.aws_conn_id, region_name=self.region).async_conn as logs_client:
+            # fmt: on
+            waiter = ecs_client.get_waiter("tasks_stopped")
+            logs_token = None
+            while True:
+                try:
+                    await waiter.wait(
+                        cluster=self.cluster, tasks=[self.task_arn], WaiterConfig={"MaxAttempts": 1}
+                    )
+                    break  # we reach this point only if the waiter met a success criteria
+                except WaiterError as error:
+                    if "terminal failure" in str(error):
+                        raise
+                    self.log.info("Status of the task is %s", error.last_response["tasks"][0]["lastStatus"])
+                    await asyncio.sleep(int(self.waiter_delay))
+                finally:
+                    if self.log_group and self.log_stream:
+                        logs_token = await self._forward_logs(logs_client, logs_token)
+
+        yield TriggerEvent({"status": "success", "task_arn": self.task_arn})
+
+    async def _forward_logs(self, logs_client, next_token: str | None = None) -> str | None:
```

Comment:
this is code that _very inspired_ by https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/hooks/logs.py#L53 but since I need to use an async call in the middle, refactoring the existing code to allow that seemed like a lot of added complexity to the existing code

Reply:
should this be part of the hook implementation?

Reply:
of the logs hook you mean ? idk, I wasn't sure if it was a good idea to duplicate the code in the hook, I found it easier to write something fitting exactly my need here.